File: examples/typescript/basic_chat.ts
// basic_chat.ts — Fully typed DeepSeek chat completion (TypeScript)
//
// Usage:
//   export DEEPSEEK_API_KEY="sk-your-key"
//   npm install openai tsx
//   npx tsx basic_chat.ts

import OpenAI from "openai";

const client = new OpenAI({
  apiKey: process.env.DEEPSEEK_API_KEY!,
  baseURL: "https://api.deepseek.com",
});

interface ChatOptions {
  model?: string;
  systemPrompt?: string;
  maxTokens?: number;
}

async function chat(
  userMessage: string,
  options: ChatOptions = {}
): Promise<string> {
  const {
    model = "deepseek-flash",
    systemPrompt = "You are a helpful assistant.",
    maxTokens = 1024,
  } = options;

  const response = await client.chat.completions.create({
    model,
    messages: [
      { role: "system", content: systemPrompt },
      { role: "user", content: userMessage },
    ],
    max_tokens: maxTokens,
  });

  const content = response.choices[0].message.content;
  if (!content) throw new Error("Empty response from model");

  return content;
}

async function main(): Promise<void> {
  const result = await chat(
    "Write a one-line TypeScript function that reverses a string.",
    {
      model: "deepseek-flash",
      systemPrompt: "You are an expert TypeScript developer. Be concise.",
    }
  );

  console.log("Answer:", result);
}

main().catch(console.error);

File: examples/python/basic_chat.py
"""
basic_chat.py — Minimal DeepSeek chat completion example.

Usage:
    export DEEPSEEK_API_KEY="sk-your-key"
    python basic_chat.py
"""

import os
from openai import OpenAI


def main():
    client = OpenAI(
        api_key=os.environ.get("DEEPSEEK_API_KEY"),
        base_url="https://api.deepseek.com",
    )

    response = client.chat.completions.create(
        model="deepseek-flash",  # Use deepseek-v4-pro for more complex tasks
        messages=[
            {"role": "system", "content": "You are a helpful assistant."},
            {"role": "user", "content": "What is the capital of France?"},
        ],
        max_tokens=256,
    )

    print(response.choices[0].message.content)
    print(f"\n[Usage] prompt={response.usage.prompt_tokens}, "
          f"completion={response.usage.completion_tokens}, "
          f"total={response.usage.total_tokens}")
    print(f"[Cache] hits={response.usage.prompt_cache_hit_tokens}, "
          f"misses={response.usage.prompt_cache_miss_tokens}")


if __name__ == "__main__":
    main()
